lib/src/analyzers/lint_analyzer/rules/rules_list/ban_name/ban_name_rule.dart

```diff
@@ -33,11 +33,32 @@ class BanNameRule extends CommonRule {
 
     source.unit.visitChildren(visitor);
 
-    return visitor.nodes
+    final filteredNodeList = visitor.nodes.where((node) {
+      if (node.endNode != null) {
```

Comment:
Could you elaborate, what you're actually trying to check here?

Reply:
The visitor finds 2 adjacent nodes separated by a single character. This character seems to be only checkable inside a rule based on source.content
Without a filter, the last line in the test will be lint when it shouldn't be.

This filter trying to find an ident in range between `node.offset` and `endNode.end` inside a `source.content`
For ex.:
```
// analysis_options.yaml
... 
- ban-name:
     entries:
       - ident: DateTime.now
...

// code
DateTime now = DateTime(2022);
```
1. Without this filter, the visitor finds a `DateTime now` because it doesn't know anything about the separator
2. Filter checks if "node" `DateTime now = DateTime(2022);` contains an `DateTime.now` ident

Reply:
Okay, I've cloned the implementation and tested it locally. 

I think the approach should be changed.
Instead of trying to build a string from identifier names, let's try other thing - let's check for node parent and if there is a method invocation, variable access or property access, we will extract the "path" from parent and check if it exists in the config.

So, if I add to the test `DateTime.now().day.isFinite;` and visit `isFinite`, the parent property will be the following:
<img width="968" alt="Screenshot 2022-10-16 at 13 34 56" src="https://user-images.githubusercontent.com/21116742/196028467-b4b63da5-993a-41d7-9071-7bb97469c62f.png">

Changing the approach will allow to remove this check and check everything inside the visitor, plus will not rely on the string representation, but rather on parsed AST.